os_migrate/plugins/module_utils/resource.py

```diff
@@ -0,0 +1,198 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from ansible_collections.os_migrate.os_migrate.plugins.module_utils import const
+from ansible_collections.os_migrate.os_migrate.plugins.module_utils import exc
+
+
+class Resource():
+
+    resource_type = 'UNDEFINED'
+    sdk_class = 'UNDEFINED'
+
+    # Serialization
+    info_from_sdk = []
+    info_from_refs = []
+    params_from_sdk = []
+    params_from_refs = []
+
+    # Deserialization
+    sdk_params_from_params = []
```

Comment:
As discussed in #105 this is probably YAGNI and we can use `params_from_sdk` in its place.